lib/workspace.js

```diff
@@ -336,17 +337,53 @@ module.exports = {
     // local workspace files
     const questionBasePath = path.join(course_path, 'questions', question.qid);
     const localPath = path.join(questionBasePath, 'workspace');
+    const templatePath = path.join(questionBasePath, 'workspaceTemplates');
 
     // base workspace directory wherever we are uploading to
     const remoteDirName = `workspace-${workspace_id}-${workspace.version}`;
     const remotePath = path.join(remoteDirName, 'current');
 
-    // Check if we have any local files to upload
-    const localPathExists = await fse.pathExists(localPath);
+    const staticFiles = (
+      await async
+        .mapSeries(klaw(localPath), async (file) => {
+          return file.stats.isFile()
+            ? { name: path.relative(localPath, file.path), localPath: file.path }
+            : null;
+        })
+        .catch(() => {
+          // Template path does not exist or is not accessible, do nothing
+          return [];
+        })
+    ).filter((file) => file);
+
+    const mustacheParams = { params: variant.params, correct_answers: variant.true_answer };
+
+    const templateFiles = (
+      await async
+        .mapSeries(klaw(templatePath), async (file) => {
+          try {
+            if (!file.stats.isFile()) return null;
+            return {
+              name: path.relative(templatePath, file.path),
+              buffer: mustache.render(
+                await fsPromises.readFile(file.path, { encoding: 'utf-8' }),
+                mustacheParams
+              ),
+            };
+          } catch (_err) {
+            // File cannot be rendered, treat file as static file
```

Comment:
I'm thinking we should throw an error instead here? It feels better to break the question if something is not working correctly rather than to continue and hope for the best.

Reply:
I'm considering cases here of binary files that may be included in the templates by the instructor for some reason, but I'm happy to remove this if necessary. In any case, none of the errors so far cause an issue to be created, though there was discussion in the previous PR to do this in the future.

Reply:
> In any case, none of the errors so far cause an issue to be created, though there was discussion in the previous PR to do this in the future.

See #7095.

Reply:
Excellent, I agree we should be creating issues in these cases.

Reply:
I also think that it's fine for binary files to create errors -- they should be moved to `workspace/`. I really don't want to have silent failures during question creation.

Reply:
These errors actually occur during *workspace* creation, which happen outside of the question generation flow. As there's no request/response flow, there's also no way for us to surface meaningful errors to students or instructors. Ideally we'd both create an issue *and* show a meaningful error directly to the user (I know that's outside the scope of this PR though).

Reply:
Good point! One option would be to create the issue and then set the workspace state to `stopped` with a message that reflects the error. This will be immediately shown to the student.

Reply:
Moved these comments to #7095 so they're not lost.